packages/react/src/SelectPanel/SelectPanel.tsx

```diff
@@ -328,40 +340,106 @@ export function SelectPanel({
+              return
+            }
 
-          if (isMultiSelectVariant(selected)) {
-            const otherSelectedItems = selected.filter(selectedItem => !areItemsEqual(selectedItem, item))
-            const newSelectedItems = doesItemsIncludeItem(selected, item)
-              ? otherSelectedItems
-              : [...otherSelectedItems, item]
+            if (isMultiSelectVariant(selected)) {
+              const otherSelectedItems = selected.filter(selectedItem => !areItemsEqual(selectedItem, item))
+              const newSelectedItems = doesItemsIncludeItem(selected, item)
+                ? otherSelectedItems
+                : [...otherSelectedItems, item]
 
-            const multiSelectOnChange = onSelectedChange as SelectPanelMultiSelection['onSelectedChange']
-            multiSelectOnChange(newSelectedItems)
-            return
-          }
+              const multiSelectOnChange = onSelectedChange as SelectPanelMultiSelection['onSelectedChange']
+              multiSelectOnChange(newSelectedItems)
+              return
+            }
 
-          // single select
-          const singleSelectOnChange = onSelectedChange as SelectPanelSingleSelection['onSelectedChange']
-          singleSelectOnChange(item === selected ? undefined : item)
-          onClose('selection')
-        },
-      } as ItemProps
-    })
-  }, [onClose, onSelectedChange, items, selected])
+            // single select
+            const singleSelectOnChange = onSelectedChange as SelectPanelSingleSelection['onSelectedChange']
+            singleSelectOnChange(item === selected ? undefined : item)
+            onClose('selection')
+          },
+        } as ItemProps
+      })
+      .sort((itemA, itemB) => {
+        // itemA is selected (for sorting purposes) if an object in selectedOnSort matches every property of itemA, except for the selected property
+        const itemASelected = selectedOnSort.some(item =>
+          Object.entries(item).every(([key, value]) => {
+            if (key === 'selected') {
+              return true
+            }
+            return itemA[key as keyof ItemProps] === value
+          }),
+        )
+
+        // itemB is selected (for sorting purposes) if an object in selectedOnSort matches every property of itemA, except for the selected property
+        const itemBSelected = selectedOnSort.some(item =>
+          Object.entries(item).every(([key, value]) => {
+            if (key === 'selected') {
+              return true
+            }
+            return itemB[key as keyof ItemProps] === value
+          }),
+        )
+
+        // order selected items first
+        if (itemASelected > itemBSelected) {
+          return -1
+        } else if (itemASelected < itemBSelected) {
+          return 1
+        }
+
+        // Then order by text
+        if ((itemA.text ?? '') < (itemB.text ?? '')) {
+          return -1
+        } else if ((itemA.text ?? '') > (itemB.text ?? '')) {
+          return 1
+        }
```

Comment:
I baked in alphabetically ordering through text but now that I'm looking at the [issue](https://github.com/github/primer/issues/2409) it doesn't really look like a requirement 🤔 
I can remove and do selection-at-top only if this is not wanted/needed 👀 
CC: @emilybrick

Reply:
Consumers could want to sub-sort based on something besides alphabetical order. For example: sort most frequently used labels to the top.

I think we should remove this default alphanumeric sort.

Reply:
Feel free to steal the alphanumeric stuff from data table btw if it's helpful! We have some sort helpers over there